### PR TITLE
Updates the Monitoring Service Spec according to current design

### PR DESCRIPTION
**Pull request by Dominik1999**

### CODE REVIEW

File: monitoring_service.rst [hunk 1]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
-* Good enough uptime (a third party service to monitor the servers can be used to provide statistics)
-* Sybil Attack resistance (i.e. no one should be able to announce an unlimited number of (faulty) services)
-* Some degree of redundancy (ability to register balance proof with multiple competing monitoring services)
-* A stable and fast ethereum node connection (channel update transactions should be propagated ASAP as there is competition among the monitoring services)
-* If MS registry is used, a deposit will be required for registration
+Summary
+=======
+*   Monitoring Service (MS) listens to blinded Balance Proofs (BP) and proposed service fees in a public Matrix Room
+*   Any pre-registered MS can decide to monitor a channel and store corresponding BPs
+*   Then, whenever a channel is closed by calling `closeChannel` the MS will call `updateNonClosingBalanceProof` with
```
Comment (palango) [inline monitoring_service.rst]: It should be added that the client is expected to react first, so it gets a period to update itself.

File: monitoring_service.rst [hunk 2]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+The Raiden node that belongs to Alice is going offline and Alice wants to be protected against having her channels
+closed by Bob with an incorrect `balance proof`.
 
-1) Alice broadcasts the balance proof by sending a message to a public chat room.
-2) Monitoring services decide if the fee is worth it and picks the balance proof up.
-3) Alice now goes offline.
-4) Bob sends an on-chain transaction in attempt to use an earlier balance proof that is in his favor.
-5) Some of the monitoring servers detect that an incorrect :term:`BP` is being used. They can update the channel closing state as long as the :term:`Challenge Period` is not over.
-6) After the Challenge period expires, the channel can be settled with a balance that Alice expects to be correct.
+1)  Whenever Alice wants to get her channel monitored, Alice publishes her Monitor Request - blinded balance proofs -
```
Comment (palango) [inline monitoring_service.rst]: The MR includes a blinded balance proof, that should be made clear. Also we can link this to the MR definition.

File: monitoring_service.rst [hunk 3]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+    into a public chat room
+2)  Updates can be sent whenever the state or proposed fee changes
```
Comment (palango) [inline monitoring_service.rst]: can be sent -> can be sent by Alice

File: monitoring_service.rst [hunk 4]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+3)  MSs can pick up these messages and then listen to closeChannel events regarding this particular channel
```
Comment (palango) [inline monitoring_service.rst]: The event is called `ChannelClosed`, and please mark it as code.

File: monitoring_service.rst [hunk 5]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+4)  If so, whenever the channel is closed and before the settlement period has ended, the MS will call the
+    `updateNonClosingBalanceProof` function with the provided information. This way the settleChannel function that
+    calculates the token distribution can only be executed submitting the corresponding balance values of the hashed
+    balances provided before
```
Comment (palango) [inline monitoring_service.rst]: Pls add that the client gets time to react first.

File: monitoring_service.rst [hunk 6]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+5)  Either Alice or Bob (or anyone else) can now anytime call settleChannel and initiate the token distribution
```
Comment (palango) [inline monitoring_service.rst]: Mark `settleChannel` as code

File: monitoring_service.rst [hunk 7]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+6)  This will result in a correct distribution of tokens, in accordance with the off-chain transfers
```
Comment (palango) [inline monitoring_service.rst]: This can be put into the last point.

File: monitoring_service.rst [hunk 8]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+7)  After a successful submission of the hashed balance by the MS (might be a defeat of an attack), the monitoring
+    service gets an on-chain payment from a smart contract where Alice deposited some tokens beforehand
```
Comment (palango) [inline monitoring_service.rst]: I'd also mention which function in the SC is used for this.

File: monitoring_service.rst [hunk 9]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+also publish their expected rewards in this room, which does not provide any guarantees, but increases the chance of
+reliable monitoring if both parties cooperate.
 
-Once it does accept a BP it MUST provide monitoring for the associated channel at least until a newer BP is provided or the channel is settled. MS SHOULD continue to accept newer balance proofs for the same channel.
+Monitoring Service Payment
+--------------------------
 
-Once a `ChannelClosed` or `NonClosingBalanceProofUpdated` event is seen the MS MUST verify that the channel’s balance matches the latest BP it accepted. If the balances do not match the MS MUST submit that BP to the channel’s `updateNonClosingBalanceProof` method.
+The MS is paid after successfully submitting its client’s balance proof update. The payment is paid out from a deposit
+in the User Deposit Contract (UDC). Ideally, only one MS submits the latest BP to the SC to avoid unnecessary gas usage.
```
Comment (palango) [inline monitoring_service.rst]: Can you add a note that this is still WIP

File: monitoring_service.rst [hunk 10]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+requests. Since there is no punishment for failing to monitor a channel, stopping to monitor is a simple way to reduce
+resource usage when desired, although it should not be necessary under normal circumstances.
 
-How will raiden nodes specify/deposit the monitoring fee? How will it be collected?
+Proposed SC logic
+'''''''''''''''''
 
-A scheme to prevent unnecessary simultaneous updates needs to exist. Options:
-MS chose an order amongst themselves
+1)  Raiden node will transfer tokens used as a reward to the channel smart contract
```
Comment (palango) [inline monitoring_service.rst]: Is this correct? The token are already in the UDC, right?

File: monitoring_service.rst [hunk 11]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+2)  Whoever calls SC’s updateTransfer method MUST supply payout address as a parameter. This address is stored in the SC
```
Comment (palango) [inline monitoring_service.rst]: please mark the function names as code. There's also a period missing after *This address is stored in the SC*, also add which SC is meant.

File: monitoring_service.rst [hunk 12]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+    updateTransfer MAY be called multiple times, but it will only accept BP newer than the previous one
```
Comment (palango) [inline monitoring_service.rst]: *a* BP

File: monitoring_service.rst [hunk 13]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+3)  When settling (calling contract suicide), the reward tokens will be sent to the payout address
```
Comment (palango) [inline monitoring_service.rst]: The MS needs to call `claimReward` to get the reward.

File: monitoring_service.rst [hunk 14]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
 Appendix A: Interfaces
 ======================
 
-Broadcast interface
+Broadcast Interface
 -------------------
-Client's request to store Balance Proof will be in the usual scenario broadcasted using Matrix as a transport layer. A public chatroom will be available for anyone to join - clients will post balance proofs to the chatroom and Monitoring Service picks them up.
+Client's request to store a `balance proof` will be in the usual scenario broadcasted using Matrix as a transport layer.
+A public chatroom will be available for anyone to join - clients will post balance proofs to the chatroom and
+Monitoring Service picks them up.
```
Comment (palango) [inline monitoring_service.rst]: Service*s*

File: monitoring_service.rst [hunk 15]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
+|  reward_amount           | uint256    | Proposed fee in RDN                                                            |
++--------------------------+------------+--------------------------------------------------------------------------------+
+|  reward_proof_signature  | string     | Signature of parts of the Onchain Balance Proof - channel_identifier,          |
+|                          |            | token_network_address, chain_id, nonce - and the reward_amount                 |
++--------------------------+------------+--------------------------------------------------------------------------------+
+
+__ https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#balance-proof-onchain
 
-All of this fields are required. Monitoring Service MUST perform basic verification of these data, namely channel existence. Monitoring service SHOULD accept the message if and only the sender of the message is same as the sender address recovered from the signature.
+All of this fields are required. Monitoring Service MUST perform basic verification of these data, namely channel
```
Comment (palango) [inline monitoring_service.rst]: what is *basic* verification? I'd just remove that word.

File: monitoring_service.rst [hunk 16]
```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
...
 
     {
-      "nonce": 13,
-      "transferred_amount": 15000,
-      "channel_address": "0x87F5636c67f2Fd4F11710974766a5B1b6f33FB1d",
-      "extra_hash": "0xe0fa3e376941dafc9b3836f80bee307ab2eacb569ec7ccceff5e66b48b1efd9c",
-      "locksroot": "0xebd7dc7d6dd7956e62104182194939a1223c738ffc2a14dbbecb6191cf76f211",
-      "signature": "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+      "onchain_balance_proof": {
+                                 "signature_prefix": tbd,
```
Comment (palango) [inline monitoring_service.rst]: No need for so much indention. Just indent it 4 spaces.

File: monitoring_service.rst (hunk 1 above)
Comment (Dominik1999) [reply monitoring_service.rst]: good point I will add "after x blocks" and we can include the number of blocks as soon we have decided

File: monitoring_service.rst (hunk 2 above)
Comment (Dominik1999) [reply monitoring_service.rst]: added

File: monitoring_service.rst (hunk 3 above)
Comment (Dominik1999) [reply monitoring_service.rst]: done

File: monitoring_service.rst (hunk 5 above)
Comment (Dominik1999) [reply monitoring_service.rst]: done

File: monitoring_service.rst (hunk 7 above)
Comment (Dominik1999) [reply monitoring_service.rst]: good point

File: monitoring_service.rst (hunk 8 above)
Comment (Dominik1999) [reply monitoring_service.rst]: done

File: monitoring_service.rst (hunk 9 above)
Comment (Dominik1999) [reply monitoring_service.rst]: yes

File: monitoring_service.rst (hunk 10 above)
Comment (Dominik1999) [reply monitoring_service.rst]: yes good catch

File: monitoring_service.rst (hunk 11 above)
Comment (Dominik1999) [reply monitoring_service.rst]: done

File: monitoring_service.rst (hunk 15 above)
Comment (Dominik1999) [reply monitoring_service.rst]: good point

File: monitoring_service.rst (hunk 16 above)
Comment (Dominik1999) [reply monitoring_service.rst]: thank you for the detailed feedback, everything is done

Comment (palango) [reply monitoring_service.rst]: the closing bracecan be unindented as well

File: monitoring_service.rst [hunk 17]
```diff
@@ -2,145 +2,234 @@ Raiden Network Monitoring Service
...
-* Some degree of redundancy (ability to register balance proof with multiple competing monitoring services)
-* A stable and fast ethereum node connection (channel update transactions should be propagated ASAP as there is competition among the monitoring services)
-* If MS registry is used, a deposit will be required for registration
-
-Usual scenario
+Summary
+=======
+*   Monitoring Service (MS) listens to blinded Balance Proofs (BP) and proposed service fees in a public Matrix Room
+*   Any pre-registered MS can decide to monitor a channel and store corresponding BPs
+*   Then, whenever a channel is closed by calling `closeChannel` and after a period of x blocks where the client is
```
Comment (palango) [inline monitoring_service.rst]: This doesn't appear as code, I guess you need double *'*

**Pull request merged**